### PR TITLE
Add workload identity

### PR DESCRIPTION
**Pull request by PheonixS**
### Added
Kubernetes Workload Identity from azidentity library.
### Changed
Bumped version of azidentity from v1.2.0 to v1.3.0

### CODE REVIEW

**Comment by PheonixS** [comment]
@microsoft-github-policy-service agree

**Review by OlehPriadka-TomTom — APPROVED** [review]

**Review by OlehPriadka-TomTom — CHANGES_REQUESTED** [review]
approve by mistake

**Comment by AsafMah** [comment]
WithKubernetesWorkloadIdentity() should accept the parameters it needs, instead of assuming they are in the string.
If the parameters are optional - just mention it in the docs.

File: kusto/kcsb.go [hunk 1]
```diff
@@ -251,6 +254,15 @@ func (kcsb *ConnectionStringBuilder) WithSystemManagedIdentity() *ConnectionStri
 	return kcsb
 }
 
+// WithKubernetesWorkloadIdentity Creates a Kusto Connection string builder that will authenticate with AAD application, using
+// an application token obtained from a Microsoft Service Identity endpoint using Kubernetes workload identity.
+func (kcsb *ConnectionStringBuilder) WithKubernetesWorkloadIdentity() *ConnectionStringBuilder {
```
Comment (yogilad) [inline kusto/kcsb.go:259]: The API ought to take in any argument that is required to configure this auth mode. 
This one seems to require 3 (app id, token file, authority id)

File: kusto/kcsb.go [hunk 2]
```diff
@@ -23,6 +24,8 @@ type ConnectionStringBuilder struct {
 	ApplicationToken                 string
 	AzCli                            bool
 	MsiAuthentication                bool
+	WorkloadAuthentication           bool
+	FederationTokenFile              string
```
Comment (yogilad) [inline kusto/kcsb.go:28]: Is this the file content or the file path?
if the latter, please add "Path" to the member name.

File: kusto/kcsb_test.go [hunk 3]
```diff
@@ -166,6 +166,12 @@ func TestGetTokenProviderHappy(t *testing.T) {
 				DataSource:        "https://endpoint/test_tokenprovider_managedidauth2",
 				MsiAuthentication: true,
 			},
+		}, {
```
Comment (yogilad) [inline kusto/kcsb_test.go:169]: Please also add an API test for WithKubernetesWorkloadIdentity(...)

File: kusto/kcsb.go (hunk 1 above)
Comment (yogilad) [reply kusto/kcsb.go:259]: This ought to also affect tests and code example.

**Review by yogilad — COMMENTED** [review]
Thanks for the contribution.
I left comments in the PR.

File: kusto/kcsb.go (hunk 2 above)
Comment (PheonixS) [reply kusto/kcsb.go]: done

File: kusto/kcsb_test.go (hunk 3 above)
Comment (PheonixS) [reply kusto/kcsb_test.go:183]: done

File: kusto/kcsb.go (hunk 1 above)
Comment (PheonixS) [reply kusto/kcsb.go]: done

**Review by AsafMah — APPROVED** [review]

**Pull request merged**